src/main/java/io/strimzi/kafka/bridge/http/HttpBridge.java

```diff
@@ -208,11 +209,17 @@ private void sendToPartition(RoutingContext routingContext) {
     private void createConsumer(RoutingContext routingContext) {
         this.httpBridgeContext.setOpenApiOperation(HttpOpenApiOperations.CREATE_CONSUMER);
 
-        JsonObject body = routingContext.getBodyAsJson();
+        JsonObject bodyAsJson = null;
+        // TODO: it seems that getBodyAsJson raises an exception when the body is empty and not null
+        try {
+            bodyAsJson = routingContext.getBodyAsJson();
+        } catch (DecodeException ex) {
```

Comment:
I think this is dangerous, because I'm guessing `DecodeException` can be thrown in other circumstances (such as malformed JSON). So you should get the body in some other representation (is there a `getBodyAsString()`?) where you can check for the nullness of the body prior to calling `getBodyAsJson()`.

Also if you assign `bodyAsJson` an empty object rather than null in the case that the body is null you won't need to futz around adding `JsonObject requestBody` in `HttpSinkBridgeEndpoint.doCreateConsumer()`.

Reply:
It's how it works even in the http sink and source but I see your point.

> Also if you assign bodyAsJson an empty object rather than null in the case that the body is null you won't need to futz around adding JsonObject requestBody in HttpSinkBridgeEndpoint.doCreateConsumer().

That was my first implementation but I was actually changing the "original" body which was something I didn't like. At the same time I didn't like to change the other env vars of course but actually changing the original one is something that could drive to collateral effects in general, maybe not in this case. I was trying to be a good developer :-)

Reply:
You're not really "changing the original body", you're treating no body an synonymous with an empty (JSON) object, which doesn't care about the distinction. If it did care, then obviously you'd need to reflect that somehow. But another reason is the fewer nulls you pass around the less likely you are to be hit by an NPE due to a missing null check.